source/material_model/rheology/strain_dependent.cc

```diff
@@ -501,33 +501,52 @@ namespace aspect
             const double edot_ii = std::max(sqrt(std::fabs(second_invariant(deviator(in.strain_rate[i])))),min_strain_rate);
             double delta_e_ii = edot_ii*this->get_timestep();
 
-            // Adjusting strain values to account for strain healing without exceeding an unreasonable range
-            if (healing_mechanism != no_healing)
+            // Assign accumulated strain value according to active deformation mechanism
+
+            // Plastic strain
+            double delta_e_ii_plastic = 0.;
+            if (plastic_yielding == true)
+              delta_e_ii_plastic = delta_e_ii;
+
+            // Viscous strain
+            double delta_e_ii_viscous = 0.;
+            if (plastic_yielding == false)
+              delta_e_ii_viscous = delta_e_ii;
+
+            // Total strain
+            double delta_e_ii_total = delta_e_ii;
+
+            // Now account for strain healing
+            if (healing_mechanism == temperature_dependent)
```

Comment:
Keep the if-condition, otherwise other healing mechanisms in the future will not work.
```suggestion
            if (healing_mechanism != no_healing)
```